client/settings/survey-modal/provider.js

```diff
@@ -10,41 +10,33 @@ import apiFetch from '@wordpress/api-fetch';
 import WcPaySurveyContext from './context';
 import { NAMESPACE } from '../../data/constants';
 
-// @todo - prepare data from form and pass to data in POST method.
-
 const WcPaySurveyContextProvider = ( { children } ) => {
-	const [ isSurveySubmitted, setSurveySubmitted ] = useState(
-		Boolean( false )
-	);
+	const [ isSurveySubmitted, setSurveySubmitted ] = useState( false );
 	const [ status, setStatus ] = useState( 'resolved' );
 	const [ surveyAnswers, setSurveyAnswers ] = useState( {} );
 
-	const submitSurvey = useCallback(
-		( value ) => {
```

Comment:
The survey data is already part of the component, so there's no need to pass it as an argument to `submitSurvey`. `submitSurvey` can just consume `surveyAnswers`.